### PR TITLE
Add clang compilation option from kernel 4.19

### PR DESCRIPTION
**Pull request by vaibhav-rustagi**
Changes for choosing compiler based on kernel versions.

### CODE REVIEW

File: cos-gpu-installer-docker/Dockerfile [hunk 1]
```diff
@@ -19,11 +19,8 @@ LABEL maintainer="cos-containers@google.com"
 
 # Install minimal tools needed to build kernel modules.
 RUN apt-get update -qq && \
-    apt-get install -y xz-utils python2.7-minimal kmod git make bc curl ccache libc6-dev pciutils gcc libelf-dev libssl-dev && \
-    rm -rf /var/lib/apt/lists/*
-
-# Download & install prebuild COS toolchain package, and prepare the environment for cross-compiling.
-RUN cd /usr/bin && ln -s python2.7 python && ln -s python2.7 python2
+    apt-get install -y xz-utils python2.7-minimal kmod git make bc curl ccache libc6-dev pciutils \
```
Comment (kewu1992) [inline cos-gpu-installer-docker/Dockerfile]: Why add the slash? If you plan to format it, wrap it at 80 characters.

File: cos-gpu-installer-docker/entrypoint.sh [hunk 2]
```diff
@@ -214,7 +214,12 @@ install_cross_toolchain_pkg() {
   info "Configuring environment variables for cross-compilation"
   export PATH="/build/bin:${PATH}"
   export SYSROOT="/build/usr/x86_64-cros-linux-gnu"
-  export CC="x86_64-cros-linux-gnu-gcc"
+  local kernel_version="$(uname -r | grep -o "^[0-9]*\.[0-9]*" | head -1)"
+  if [[ "${kernel_version}" = 4.14 ]]; then
```
Comment (kewu1992) [inline cos-gpu-installer-docker/entrypoint.sh]: Please also include 4.4. Although all 4.4 images are deprecated, we want the cos-gpu-installer to be backward compatible. 

Also add a comment to indicate COS-kernel will be compiled by clang starting from 4.19.

**Review by kewu1992 — CHANGES_REQUESTED** [review]

File: cos-gpu-installer-docker/entrypoint.sh (hunk 2 above)
Comment (adityakali) [reply cos-gpu-installer-docker/entrypoint.sh]: I feel like this should not be kernel version based, but COS image version based. For example, what if we roll-back CLANG for 4.19? Or rollout a 4.14 based image with CLANG?
Also, the indication whether which compiler is used for a particular image version should come from either the image itself (ex: `/etc/toolchain`) or from the corresponding artifacts in `gs://cos-tools/<build#>`.

Comment (SaiedKazemi) [reply cos-gpu-installer-docker/entrypoint.sh]: I agree. We should not determine the toolchain used to build the kernel from the kernel version and/or name.

**Pull request closed without merging**